snapcraft/remote/git.py

```diff
@@ -84,7 +84,7 @@ def get_git_repo_type(path: Path) -> GitType:
     :returns: GitType
     """
     if is_repo(path):
-        repo = pygit2.Repository(path)
+        repo = pygit2.Repository(path)  # type: ignore[reportAttributeAccessIssue]
```

Comment:
I couldn't find any better alternative than to ignore this.  (I tested craft-application, the same issue will occur once it gets a pyright update)

Reply:
Maybe hold pyright for now?

Reply:
Perhaps.  I don't think it's a bug but rather pyright being more precise with stricter checks.
I would like to get @lengau opinion too.

Reply:
pyright 1.1.356 fixed that

Reply:
Ooo thank you, I hadn't noticed!  I'll give that a try

Reply:
Yep, that fixed it.  I should have tried 1.1.356 from the start.